app/src/main/java/org/mozilla/fenix/compose/tabstray/MediaImage.kt

```diff
@@ -0,0 +1,83 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.compose.tabstray
+
+import androidx.compose.foundation.border
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.layout.width
+import androidx.compose.foundation.shape.CircleShape
+import androidx.compose.material.Card
+import androidx.compose.material.Icon
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import mozilla.components.browser.state.state.TabSessionState
+import mozilla.components.browser.state.state.createTab
+import mozilla.components.concept.engine.mediasession.MediaSession.PlaybackState
+import org.mozilla.fenix.R
+import org.mozilla.fenix.theme.FirefoxTheme
+
+/**
+ * Draws controller buttons for the media (play/pause) state of the given [tab].
+ *
+ * @param tab the tab which the image should be shown.
+ * @param onMediaIconClicked handles the click event when tab has media session like play/pause.
+ * @param modifier [Modifier] to be applied to the layout.
+ */
+@Composable
+fun MediaImage(
+    tab: TabSessionState,
+    onMediaIconClicked: ((TabSessionState) -> Unit),
+    modifier: Modifier,
+) {
+    val (icon, contentDescription) = when (tab.mediaSessionState?.playbackState) {
+        PlaybackState.PAUSED -> {
+            R.drawable.media_state_play_vector to R.string.mozac_feature_media_notification_action_play
+        }
+        PlaybackState.PLAYING -> {
+            R.drawable.media_state_pause_vector to R.string.mozac_feature_media_notification_action_pause
+        }
+        else -> return
+    }
+
+    Card(
+        modifier = modifier
+            .size(size = 24.dp)
+            .clip(shape = CircleShape)
+            .border(
+                width = 2.dp,
+                color = FirefoxTheme.colors.layer2,
+                shape = CircleShape
+            ),
+        backgroundColor = FirefoxTheme.colors.layerAccent
+    ) {
+        Icon(
+            painter = painterResource(id = icon),
+            modifier = Modifier
+                .size(size = 24.dp)
+                .clickable { onMediaIconClicked(tab) },
```

Comment:
I think we're trying to use `IconButton` instead of an `Icon` + `Modifier.clickable`. Alternatively, the Card has a built-in onClick parameter if we want to have the entire card to be clickable. `IconButton` though adds some baked in padding to make the tap area more accessible.